### PR TITLE
Atexit fix

### PR DESCRIPTION
**Pull request by 3tilley**
A library we use throws this annoying error when it's called in under certain conditions. It's probably doing something wrong, but I think `colorama` should be more resilient to this particular bad state. I've attached the script that demonstrates the error. I've tried to write a test for this, but the best I can to do honestly test the functionality is to spin up a new Python in a subprocess that calls `colorama`,  this seems awkward and was difficult to make portable. If it's your desire I'll do it though.

```
from __future__ import print_function

import codecs
import sys

from colorama import init

init()
sys.stdout = codecs.getwriter("utf-8")(sys.stdout.detach())

print("foo", sys.stdout)
```

Before and after the fix:
image

image

Rejected test attempt:
```
    def testDetachAfterInitDoesntThrow(self):
        fd, name = tempfile.mkstemp(suffix=".py", text=True)
        sample_python = """
from __future__ import print_function

import codecs
import sys

from colorama import init

init()
sys.stdout = codecs.getwriter("utf-8")(sys.stdout.detach())

print("Hello", sys.stdout)
        """
        temp_script = Path(name)
        with temp_script.open(mode="w") as f:
            f.write(sample_python)
        output = subprocess.check_output(["python", str(temp_script)], universal_newlines=True)
        print(output)
        assert "buffer" not in output
```

### CODE REVIEW

**Comment by 3tilley** [comment]
Great idea, added. I tried something similar before but wasn't able to detach a StringIO - I'm clearly don't have your deep stream knowledge!

**Comment by 3tilley** [comment]
I think this is just waiting a CI run, @tartley do you mind kicking it off?

**Comment by 3tilley** [comment]
Thanks so much! I literally looked at this in an output today and thought back to this PR
